packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte

```diff
@@ -90,6 +90,19 @@
     if (inSchema(toTable, toRelate.name, originalToName)) {
       errObj.toCol = colError
     }
+    if (
+      fromPrimary &&
+      fromRelate.fieldName &&
+      plusTables.filter(table => table.name === fromTable?.name)[0]?.schema[
```

Comment:
This conditional is quite hard to look at, would it be possible to break this up?

Also, if you're looking for a single entry, generally its better to use `find` rather than `filter`.

Thinking could have:
```javascript
const fromTable = plusTables.find(table => table.name === fromTable?.name)
const toTable =  plusTables.find(table => table.name === toTable?.name)
let fromType, toType
if (fromPrimary && fromRelate.fieldName) {
  fromType = fromTable?.schema[fromPrimary]?.type
  toType = toTable?.schema[fromRelate.fieldName]?.type
}
if (fromType && toType && fromType !== toType) {
  errObj.foreign = "Column type of the foreign key must match the primary key"
}
```

Reply:
Good shout. I noticed as well that on line 120 there was already a binding var for **fromTable** and **toTable** which was handy!